astropy/coordinates/matching.py

```diff
@@ -252,6 +264,10 @@ def search_around_sky(coords1, coords2, seplimit, storekdtree='_kdtree_sky'):
     Searches for pairs of points that have an angular separation at least as
     close as a specified angle.
 
+    This is inteded for use on coordinate objects with arrays of coordinates,
+    not scalars.  For scalar coordinates, it is better to use the ``separation``
+    methods.
```

Comment:
Maybe put the requirement `not scalar` (if it stays that way) in the `Parameters` section and put a fully working example of how to get the `coords2` indices for the scalar `coords1` case in the `matchsep.rst` docs? I'm not sure it will be obvious to users how to get it from this description ... at least you'd have to mention `numpy.argmin` here as you do in the error message.